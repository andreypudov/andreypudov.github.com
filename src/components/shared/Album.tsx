import Model from '../../models/core/photographyAlbum';
import Photograph from '../../models/core/photograph';
import PhotographView from './Photograph';
import VideoView from './Video';

type AlbumProps = {
  album: Model;
};

function Album(props: AlbumProps) {
  return (
    <>
    {
      props.album.getMedia().map((media, index: number) =>
        {
          switch (Object.getPrototypeOf(media)) {
            case Photograph.prototype:
              return <PhotographView album = { props.album } index = { index } key = { index } />;
            default:
              return <VideoView album = { props.album } index = { index } title = { true } description = { true } key = { index } />;
        }})
    }
    </>
  );
}

export default Album;