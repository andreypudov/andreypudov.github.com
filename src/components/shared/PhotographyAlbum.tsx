import Album from './../../models/core/photographyAlbum';
import AlbumView from './AlbumView';
import Headline from './../shared/Headline';
import Layout from './../shared/Layout';
import { useTranslation } from 'react-i18next';

type PhotographyAlbumProps = {
   album: Album;
 };

const PhotographyAlbum = ({ album }: React.PropsWithChildren<PhotographyAlbumProps>) => {
  const { i18n } = useTranslation();

  return (
    <Layout title = { album.getTitle() } cover = { album.getCover().src } social = { album.getSocial() }>
      <div className = 'container'>
        <Headline album = { album } locale = { i18n.language } />
        <AlbumView album = { album } />
      </div>
    </Layout>
  );
}

export default PhotographyAlbum;