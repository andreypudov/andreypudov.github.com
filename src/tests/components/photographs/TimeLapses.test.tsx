import { MemoryRouter } from 'react-router-dom'
import PhotographyAlbum from './../../../components/shared/PhotographyAlbum';
import Album from './../../../models/videos/timeLapses';
import { HelmetProvider } from 'react-helmet-async';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HelmetProvider>
        <MemoryRouter>
          <PhotographyAlbum album={Album} />
        </MemoryRouter>
      </HelmetProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});