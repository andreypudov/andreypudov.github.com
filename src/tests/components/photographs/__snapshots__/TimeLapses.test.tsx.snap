// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="layout"
>
  <header>
    <div
      className="container"
    >
      <div
        className="navbar navbar-expand-md navbar-dark"
      >
        <div
          className="container-fluid"
        >
          <a
            className="navbar-brand mx-auto"
            href="/"
            onClick={[Function]}
          >
            Andrey Pudov
          </a>
          <button
            aria-controls="navbarMain"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-bs-target="#navbarMain"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse justify-content-sm-center"
            id="navbarMain"
          >
            <ul
              className="navbar-nav"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/albums"
                  onClick={[Function]}
                >
                  Albums
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/contents"
                  onClick={[Function]}
                >
                  Contents
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/resume"
                  onClick={[Function]}
                >
                  Resume
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/contact"
                  onClick={[Function]}
                >
                  Contact
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </header>
  <main>
    <div
      className="container"
    >
      <h1>
        Time Lapses
        <small
          className="text-muted"
        >
          Friday, April 14, 2017
        </small>
      </h1>
      <figure>
        <div
          className="ratio ratio-16x9 gallery-video"
        >
          <iframe
            allowFullScreen={true}
            src="https://player.vimeo.com/video/517763009"
            title="Time Lapses"
          />
        </div>
        <figcaption
          className="video-caption"
        >
          Switzerland Park

Switzerland Park is an enormous green park located on the bank of the Oka River and was founded in 1903. Today, its territory accommodates an amusement park with a panoramic wheel, a zoo, and a bunch of cafes. By the way, Switzerland is not only the name of a country but also a term referring to a landscape of wooded hills. There are at least 190 “Switzerlands” in the world, one of which is in Nizhny Novgorod. The park is 3.5-kilometer long and it makes the location perfect for promenades, sports events, and marathons, often held at the location. The Oka River looks magnificent if viewed from the hills above and adds up to the delightful view gilded by the rays of the falling sun. Sentimentalists hail Nizhny Novgorod as the capital of sunsets not for nothing.
        </figcaption>
      </figure>
      <figure>
        <div
          className="ratio ratio-16x9 gallery-video"
        >
          <iframe
            allowFullScreen={true}
            src="https://player.vimeo.com/video/517762791"
            title="Time Lapses"
          />
        </div>
        <figcaption
          className="video-caption"
        >
          Zelensky Syezd Street

Construction of the stone Kremlin of Nizhny Novgorod began in 1500 with the building of the Ivanovskaya Tower; the main work commenced in 1508 and by 1515 a grandiose building was completed. The oak walls that formed the old fortifications were destroyed by a huge fire in 1513. The two-kilometer wall was reinforced by 13 towers (one of them – Zachatskaya – was on the shore of the Volga; not preserved, but was rebuilt in 2012). This “Stone City” had a permanent garrison with solid artillery weapons. With the fall of Kazan, Nizhny Novgorod Kremlin lost its military significance, and later it housed the city and provincial authorities.
        </figcaption>
      </figure>
      <figure>
        <div
          className="ratio ratio-16x9 gallery-video"
        >
          <iframe
            allowFullScreen={true}
            src="https://player.vimeo.com/video/517762848"
            title="Time Lapses"
          />
        </div>
        <figcaption
          className="video-caption"
        >
          Monument to Maxim Gorky

Alexei Maximovich Peshkov popularly known as Maxim Gorky was a Russian writer and political activist. He was nominated five times for the Nobel Prize in Literature. Before his success as an author, he traveled widely across the Russian Empire changing jobs frequently, experiences which would later influence his writing. Gorky was active in the emerging Marxist communist movement. He publicly opposed the Tsarist regime, and for a time closely associated himself with Lenin and Bogdanov's Bolshevik wing of the party.
        </figcaption>
      </figure>
      <figure>
        <div
          className="ratio ratio-16x9 gallery-video"
        >
          <iframe
            allowFullScreen={true}
            src="https://player.vimeo.com/video/517762962"
            title="Time Lapses"
          />
        </div>
        <figcaption
          className="video-caption"
        >
          Spit of Nizhny Novgorod

The Spit is one of the main natural landmarks in the historical center of Nizhny Novgorod. It is located at the confluence of the Oka and Volga rivers. In the times of the Russian Tsardom, the Spit was the center of the Strelitzky camp of Nizhny Novgorod Uyezd, thereby receiving its name. Later the river port began to form here. 
        </figcaption>
      </figure>
      <figure>
        <div
          className="ratio ratio-16x9 gallery-video"
        >
          <iframe
            allowFullScreen={true}
            src="https://player.vimeo.com/video/517762910"
            title="Time Lapses"
          />
        </div>
        <figcaption
          className="video-caption"
        >
          The State Bank of Russia

The Nizhny Novgorod branch of the State Bank on Bolshaya Pokrovskaya Street is one of the most beautiful buildings in Nizhny Novgorod and the entire Volga region. It was built in 1913 to commemorate the 300th anniversary of the Romanovs' House. The ceiling in the building, for which the neo-Russian style was chosen, was painted by the famous artist Ivan Bilibin.
        </figcaption>
      </figure>
      <figure>
        <div
          className="ratio ratio-16x9 gallery-video"
        >
          <iframe
            allowFullScreen={true}
            src="https://player.vimeo.com/video/517762737"
            title="Time Lapses"
          />
        </div>
        <figcaption
          className="video-caption"
        >
          Bolshaya Pokrovskaya Street

Bolshaya Pokrovskaya Street is the high street in the historical center of Nizhny Novgorod and one of its oldest streets. Until 1917 it was considered a street for noblemen. Formed the main street of the city by the end of the 18th century. It is considered an analog of the Arbat in Moscow or 6-7 Lines of Vasilyevsky Island in St. Petersburg.
        </figcaption>
      </figure>
    </div>
  </main>
  <footer>
    <div
      className="container"
    >
      <hr
        className="footer-divider"
      />
      <ul
        className="piped-links float-end"
      >
        <li
          className="first piped-links-short"
        >
          <a
            href="/resume"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="first piped-links-long"
        >
          <a
            href="/resume"
            onClick={[Function]}
          >
            About Andrey Pudov
          </a>
        </li>
        <li>
          <a
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
      <ul
        className="piped-links"
      >
        <p
          className="piped-links-long"
        >
          Copyright © 2022 Andrey Pudov. All rights reserved.
        </p>
        <p
          className="piped-links-short"
        >
          © 2022 Andrey Pudov. All rights reserved.
        </p>
      </ul>
    </div>
  </footer>
</div>
`;
